src/main.ts

```diff
@@ -1596,22 +1611,29 @@ function generateCachingRpcMethod(
  * types.
  */
 function generateRpcType(options: Options): InterfaceSpec {
-  const data = TypeNames.anyType('Uint8Array');
-  let fn = FunctionSpec.create('request');
+  let rpc = InterfaceSpec.create('Rpc');
+  if (options.useContext) {
+    rpc = rpc.addTypeVariable(TypeNames.typeVariable('Context'));
+  }
+  rpc = rpc.addFunction(generateRpcRequestFn({ ...options, outputJsonClientImpl: false }));
+  if (options.outputJsonClientImpl) {
+    rpc = rpc.addFunction(generateRpcRequestFn(options));
```

Comment:
I.e. here maybe just make a new `JsonRpc` that is separate from the existing `Rpc` type...